tooling/release_download_test.sh

```diff
@@ -8,164 +8,470 @@
 #   1 - Something fundamentally wrong before we could check anything
 #   2 - GPG signature verification failed
 #   3 - SHA checksum failed
-#   4 - aarch64 detected GCC/GLIBC version not as expected
+#   4 - detected GCC/GLIBC version not as expected
 #   5 - CylconeDX validation checks failed
 #   6 - SBOM contents did not meet expectations
 # Note that if there are multiple failures the highest will be the exit code
 # If there is a non-zero exit code check the output for "ERROR:"
 # 
 # For future enhancement ideas, see https://github.com/adoptium/temurin-build/issues/3506#issuecomment-1783237963
 #
-STARTDIR="$PWD"
-TAG=${1:-$TAG}
-[ -z "$TAG" ] && echo "Usage: $0 TAG" && exit 1
-[ "$(uname -m)" != "aarch64" ] && echo "This script is hard coded to be run on Linux/aarch64 - aborting" && exit 1
-
-[ "$VERBOSE" = "true" ] && set +x
-if echo "$TAG" | grep jdk8u; then
-  MAJOR_VERSION=8
-elif echo "$TAG" | grep ^jdk-; then
-  MAJOR_VERSION=$(echo "$TAG" | cut -d- -f2 | cut -d. -f1 | cut -d\+ -f1)
-else
-  # Probably a beta with the tag starting jdkXXu
-  MAJOR_VERSION=$(echo "$TAG" | cut -d- -f1 | tr -d jdku)
-fi
 
-echo "$(date +%T) : IVT : I will be checking https://github.com/adoptium/temurin${MAJOR_VERSION}-binaries/releases/tag/$TAG"
-if [ -z "${MAJOR_VERSION}" ] || [ -z "${TAG}" ]; then
-   echo "MAJOR_VERSION or TAG undefined - aborting"
-   exit 1
-fi
+set -euo pipefail
 
-if ! curl -sS "https://api.github.com/repos/adoptium/temurin${MAJOR_VERSION}-binaries/releases" > "$WORKSPACE/jdk${MAJOR_VERSION}.txt"; then
-   echo "github API call failed - aborting"
-   exit 2
-fi
+WORKSPACE=${WORKSPACE:-"$PWD"}
+VERBOSE=false
+KEEP_STAGING=false
+SKIP_DOWNLOADING=false
+
+TAG=""
+MAJOR_VERSION=""
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Comment:
Curious - why `${BASH_SOURCE[0]}` here and `${0}` in the error message a few lines later?

Reply:
I copied this line from other scripts, should the use of ${0} in the usage be updated accordingly?

Reply:
I think we should be consistent one way or another any personally I'd stick with `$0` everywhere unless you've got a specific advantage from the BASH specific version.

Reply:
ok